### DISCUSSION TITLE
Interface with user friendly

### DISCUSSION THREAD

**Discussion by tchtang** [AvaloniaUI/Avalonia#15702] [Q&A]
I dont know what title is appropriate, isnt Visual Studio problem?

If compare WPF project, there is some feature with be more helpful
1. WPF toolbox
![image](https://github.com/AvaloniaUI/Avalonia/assets/109950893/8286fd3b-17ff-4de3-8a92-dcd98bf9db0f)

non WPF
![image](https://github.com/AvaloniaUI/Avalonia/assets/109950893/d6ac0c1f-8114-4719-ad1f-10be8526fcb9)

2. WPF control properties and event handler list
![image](https://github.com/AvaloniaUI/Avalonia/assets/109950893/a24bf350-b5bd-4a84-b5cf-d1b6cbfaf643)

non WPF, click `<Grid>`
![image](https://github.com/AvaloniaUI/Avalonia/assets/109950893/51d18bab-a5e9-49e8-a35b-7fbe798aa88c)

**Comment by tchtang** [comment]
in WPF `<Button x:Name="button1" Grid.Row="3" Height="70" Width="70" Background="Blue" Click="button1_Click" />`

mouse cursor at "button1_Click", press F12, go to source code, but non-WPF, nothing happen

**Reply by timunie** [reply]
WPF is from Microsoft directly, so they can do things we can't. Rider is better when it comes to Avalonia support, but there is no WYSIWYG editor and it is unlikely to be added soon.

**Reply by thevortexcloud** [reply]
If you have access to Resharper pressing F12 works fine. But Resharper costs money. Although it is free if you are a student.

Avalonia however does not have a designer and likely never will. Using the WPF designer was almost always a bad idea due to how it worked. The Avalonia previewer works fine for most of the cases people used the WPF designer for. It can take some getting used to, but once you are able to visualise document layouts in your head it's fairly straight forwards.

**Comment by tchtang** [comment]
the event handler list is more important, I cannot remember common event handler parameter.
always double click event handler auto create function. if no this event handler list, how to create correct event function?

**Reply by rabbitism** [reply]
This works well in Rider

**Reply by tchtang** [reply]
@rabbitism thank you

**Reply by thevortexcloud** [reply]
>  if no this event handler list, how to create correct event function?

You can always check the source code for the control. I agree it's not great though. Which is why I mentioned Resharper, as Rider is build around Resharper. Except you can use Resharper from within Visual Studio, which makes Avalonia actually pretty usable in Visual Studio.

**Reply by tchtang** [reply]
@thevortexcloud thank you

**Comment by tchtang** [comment]
would like to know isnt can add this feature as below
![image](https://github.com/AvaloniaUI/Avalonia/assets/109950893/5b8fd499-430b-4230-af39-c375c972f99a)

right click a control, show can edit style
![image](https://github.com/AvaloniaUI/Avalonia/assets/109950893/03ae8ff1-f9cb-40d6-9dab-d60de0fa97aa)

above is WPF..
